src/discof/repair/fd_repair_tile.c

```diff
@@ -355,12 +356,265 @@ repair_shred_deliver( fd_shred_t const *            shred,
 }
 
 static void
-repair_shred_deliver_fail( fd_pubkey_t const * id FD_PARAM_UNUSED,
-                           ulong               slot,
-                           uint                shred_index,
-                           void *              arg FD_PARAM_UNUSED,
-                           int                 reason ) {
-  FD_LOG_DEBUG(( "repair failed to get shred - slot: %lu, shred_index: %u, reason: %d", slot, shred_index, reason ));
+fd_repair_handle_ping( fd_repair_tile_ctx_t *  repair_tile_ctx,
+                        fd_repair_t *                 glob,
+                       fd_gossip_ping_t const *      ping,
+                       fd_gossip_peer_addr_t const * peer_addr,
+                       uint                          self_ip4_addr ) {
+  fd_repair_protocol_t protocol;
+  fd_repair_protocol_new_disc(&protocol, fd_repair_protocol_enum_pong);
+  fd_gossip_ping_t * pong = &protocol.inner.pong;
+
+  fd_hash_copy( &pong->from, glob->public_key );
+
+  /* Generate response hash token */
+  uchar pre_image[FD_PING_PRE_IMAGE_SZ];
+  memcpy( pre_image, "SOLANA_PING_PONG", 16UL );
+  memcpy( pre_image+16UL, ping->token.uc, 32UL);
+
+  /* Generate response hash token */
+  fd_sha256_hash( pre_image, FD_PING_PRE_IMAGE_SZ, &pong->token );
+
+  /* Sign it */
+  repair_signer( repair_tile_ctx, pong->signature.uc, pre_image, FD_PING_PRE_IMAGE_SZ, FD_KEYGUARD_SIGN_TYPE_SHA256_ED25519 );
+
+  fd_bincode_encode_ctx_t ctx;
+  uchar buf[1024];
+  ctx.data = buf;
+  ctx.dataend = buf + sizeof(buf);
+  FD_TEST(0 == fd_repair_protocol_encode(&protocol, &ctx));
+  ulong buflen = (ulong)((uchar*)ctx.data - buf);
+
+  repair_send_intake_packet( buf, buflen, peer_addr, self_ip4_addr, repair_tile_ctx );
```

Comment:
what i actually was thinking is this would just fill the packet inside a caller provided `buf` arg

```c
fd_repair_handle_ping( fd_repair_tile_ctx_t *  repair_tile_ctx,
                       fd_repair_t *                 glob,
                       fd_gossip_ping_t const *      ping,
                       fd_gossip_peer_addr_t const * peer_addr,
                       uint                          self_ip4_addr,
                       uchar * buf,
                       ulong * buf_sz ) {
```

and then the caller would be responsible for calling `repair_send_intake_packet` (which can probably be deleted and just directly call `send_packet`)

as opposed to invoking the function itself